pthread/Makefile

```diff
@@ -0,0 +1,50 @@
+#	$OpenBSD: Makefile,v 1.49 2017/10/15 23:40:33 guenther Exp $
```

Comment:
This Makefile looks trivial. Can we avoid using recursive make and just integrate this into the main Makefile?
While we're at it, put object files in build/ and disable some warnings for pthreads?

Reply:
In order to keep the Makefile slim we might want to put rules relating to pthreads in a separate Makefile that we include from the main one. I just don't want to be using recursive make if we don't have to.